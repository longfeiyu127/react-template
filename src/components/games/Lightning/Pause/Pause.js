import React from 'react';
import './Pause.less';

function Pause(props) {
  // console.log(props);
  // eslint-disable-next-line
  const { EngineStatus, start, pause } = props;
  return (
    <div className="g-lig-Pause">
      <div
        onClick={pause}
        // onClick={() => pause()}
        className="Pause-pause iconfont icon-pause"
        name="pause"
      />
      <div className={['Pause-main', !EngineStatus ? '' : 'hide'].join(' ')}>
        <div
          className="start iconfont icon-start"
          // onClick={() => start()}
          onClick={start}
          name="start"
        />
      </div>
    </div>
  );
}

export default Pause;
