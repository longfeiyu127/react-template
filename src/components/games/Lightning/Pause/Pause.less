@import '../../../../assets/styles/theme.less';
.g-lig-Pause{
  .Pause-main{
    position: absolute;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    z-index: @firework-zindex;
    background: @fill-mask;
    display: flex;
    justify-content: center;
    align-items: center;
    color: rgba(255, 255, 255, 0.7);
  }
  .start:before{
    font-size: 100px;
  }
  .Pause-pause{
    position: absolute;
    right: 15PX;
    top: 15Px;
    z-index: @firework-zindex;
    color: rgba(255, 255, 255, 0.6);
  }
  .Pause-pause:before{
    display: block;
    font-size: 50px;
  }
}
