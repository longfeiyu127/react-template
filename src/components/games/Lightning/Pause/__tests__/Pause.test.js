import React from 'react';
import renderer from 'react-test-renderer';
// import Enzyme, { mount } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Pause from '../Pause';

// Enzyme.configure({ adapter: new Adapter() });

describe('<Pause />', () => {
  it('component render', () => {
    const start = () => {};
    const pauseFn = () => {};
    let pause = renderer
      .create(<Pause EngineStatus start={start} pause={pauseFn} />)
      .toJSON();
    expect(pause).toMatchSnapshot();

    pause = renderer
      .create(<Pause EngineStatus={false} start={start} pause={pauseFn} />)
      .toJSON();
    expect(pause).toMatchSnapshot();
  });

  // it('mount', () => {
  //   const fn = jest.fn();
  //   // const fn2 = jest.fn();
  //   const pause = mount(<Pause EngineStatus pause={fn} start={fn} />);
  //   pause.find("[name='pause']").simulate('click');
  //   expect(fn.mock.calls.length).toBe(1);
  //   // pause.find("[name='start']").simulate('click');
  //   // expect(fn2.mock.calls.length).toBe(1);
  // });
});
