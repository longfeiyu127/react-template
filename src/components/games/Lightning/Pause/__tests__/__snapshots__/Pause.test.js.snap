// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pause /> component render 1`] = `
<div
  className="g-lig-Pause"
>
  <div
    className="Pause-pause iconfont icon-pause"
    name="pause"
    onClick={[Function]}
  />
  <div
    className="Pause-main hide"
  >
    <div
      className="start iconfont icon-start"
      name="start"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`<Pause /> component render 2`] = `
<div
  className="g-lig-Pause"
>
  <div
    className="Pause-pause iconfont icon-pause"
    name="pause"
    onClick={[Function]}
  />
  <div
    className="Pause-main "
  >
    <div
      className="start iconfont icon-start"
      name="start"
      onClick={[Function]}
    />
  </div>
</div>
`;
